torch/distributed/_functional_collectives.py

```diff
@@ -353,6 +353,20 @@ def all_to_all_single(
     return _maybe_wrap_tensor(tensor)
 
 
+def permute_tensor(self: torch.Tensor, src_dst_pairs: List[Tuple[int, int]], group: RANK_TYPES, tag: str = "") -> torch.Tensor:
+    world_size = dist.get_world_size()
```

Comment:
I think the world size should be "group" world size instead of the global world size, so we should get the world_size from the `group` argument, you might want to use `expand_group` as the another collectives